Makefile

```diff
@@ -138,9 +145,12 @@ LIBCXX_CMAKE_FLAGS = \
     -DWASI_SDK_PREFIX=$(BUILD_PREFIX) \
     --debug-trycompile
 
-    #-DCMAKE_STAGING_PREFIX= 
-
 build/libcxx.BUILT: build/llvm.BUILT build/compiler-rt.BUILT build/wasi-libc.BUILT
+	# Create an empty library so that cmake's configure checks which
+	# run the compiler to see if it produces working binaries succeed.
+	mkdir -p  $(PREFIX)/lib/clang/$(CLANG_VERSION)/lib/wasi
+	$(AR) cr $(PREFIX)/share/wasi-sysroot/lib/wasm32-wasi/libc++.a
+	$(AR) cr $(PREFIX)/share/wasi-sysroot/lib/wasm32-wasi/libc++abi.a
```

Comment:
This is a little gross..  but I don't know that right way to bootstrap here.

Reply:
Actually I think its slightly better to do `-DCMAKE_C_COMPILER_WORKS=ON` on the command line here.   Basically saying "we are bootstrapping here so skip compiler checks". 

Doing it in the wasi-sdk.cmake  globally was wrong but going it locally is not so bad IMHO.

Reply:
I like that idea. I've now updated the PR to do this, and removed all the empty-archive hacks.